cardano-api/test/Test/Cardano/Api/MetaData.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TupleSections #-}
+module Test.Cardano.Api.MetaData
+  ( tests
+  ) where
+
+import           Cardano.Prelude hiding (MetaData)
+
+import           Cardano.Api.Typed
+import           Cardano.Api.MetaData
+
+import qualified Data.ByteString.Char8 as BS
+import qualified Data.Map.Strict as Map
+import qualified Data.List as List
+import qualified Data.Text as Text
+
+import           Hedgehog (Gen, Property, discover)
+import qualified Hedgehog
+import qualified Hedgehog.Gen as Gen
+import           Hedgehog.Range (Range)
+import qualified Hedgehog.Range as Range
+
+-- This test is fragile to changes in the way Metadata is converted to/from JSON.
+prop_round_trip_json_metadatum :: Property
+prop_round_trip_json_metadatum = do
+  Hedgehog.withTests 1000 . Hedgehog.property $ do
+    md <- Hedgehog.forAll genMetaData
+    Hedgehog.tripping md jsonFromMetadata jsonToMetadata
+
+-- -----------------------------------------------------------------------------
+
+-- Generate 'TxMetadata' that will round trip correctly.
+-- The only valid value that is known to not round trip correctly is a list of
+-- lists where all the sublists are of length 2.
+-- The reason this does not round trip is because a list of pairs aliases with
+-- a map.
+genMetaData :: Gen TxMetadata
+genMetaData = do
+  idxs <- List.nub <$> Gen.list (Range.linear 1 5) (Gen.word64 $ Range.linear 0 999999999999)
```

Comment:
What's the significance of `999999999999` and why don't we test the entire `Word64` range?

Reply:
Is it invalid for the list to be empty?  What should happen in this case?

Reply:
Generating `Word64` in hedgehog is a pain in the neck, mainly because `Gen.enumBounded` fails whenever the generated `Word64` is greater than `maxBound :: Int64`.

Reply:
Hmm, switched that to `Gen.word64 Range.constantBounded`